### PR TITLE
[WIP] Issue #2597 - Improving UnixSocketTest behavior.

### PR DESCRIPTION
**Pull request by joakime**
Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

### CODE REVIEW

**Comment by joakime** [comment]
@olamy and @lachlan-roberts here's the PR (and branch) for the problematic UnixSocketTest.

The changes in jetty-9.4.x have been reverted (and moved to the `jetty-9.4.x-issue-2597-unixsockettest`  branch)

File: jetty-unixsocket/src/test/java/org/eclipse/jetty/unixsocket/UnixSocketTest.java [hunk 1]
```diff
@@ -70,6 +70,8 @@ public void before() throws Exception
         testingDir.ensureEmpty();
         // Create a unique unix.sock, in target/tests directory, that's based on TestClass + testMethod + testScope
         sockFile = testingDir.getPathFile("unix.sock");
+        Files.createFile( sockFile );
+        Files.deleteIfExists( sockFile );
```
[the remark this answers is not in the document]
Comment (lachlan-roberts) [reply jetty-unixsocket/src/test/java/org/eclipse/jetty/unixsocket/UnixSocketTest.java]: why is the file being created and then immediately deleted, we know the directory is empty from line 70, so what is the point of creating and deleting

[the remark this answers is not in the document]
Comment (olamy) [reply jetty-unixsocket/src/test/java/org/eclipse/jetty/unixsocket/UnixSocketTest.java]: as the comment said it's just test to understand if we can write a file with docker build (as the error says not writable)

**Comment by olamy** [comment]
the problem is coming from filename with jenkins (/home/jenkins/workspace/jetty.project_PR-2635-K7E6LNJADYW4DDQGANLMEFBLXHJV7NNIBFPIJX6OUPTQCUYJFXNA/jetty-unixsocket/target/unix13242728516864007402.sock). jnr doesn't like. So I use a build property to use a more simple path name.

File: Jenkinsfile [hunk 2]
```diff
@@ -92,7 +92,7 @@ def getFullBuild(jdk, os) {
                       globalMavenSettingsConfig: settingsName,
                       //options: [invokerPublisher(disabled: false)],
                       mavenLocalRepo: localRepo) {
-                sh "mvn -V -B install -Dmaven.test.failure.ignore=true -e -Pmongodb -T3 -DmavenHome=${mvntoolInvoker}"
+                sh "mvn -V -B install -Dmaven.test.failure.ignore=true -e -Pmongodb -T3 -DmavenHome=${mvntoolInvoker} -Dunix.socket.tmp="+env.JENKINS_HOME
```
Comment (joakime) [inline Jenkinsfile]: Why are we polluting the JENKINS_HOME with temp files now?

Comment (olamy) [reply Jenkinsfile]: jnr cannot handle paths such /home/jenkins/workspace/jetty.project_PR-2635-K7E6LNJADYW4DDQGANLMEFBLXHJV7NNIBFPIJX6OUPTQCUYJFXNA/jetty-unixsocket/target/unix13242728516864007402.sock 
JENKINS_HOME on docker slave not master so it's not very long.
well we can use /tmp and totally skip @Before on windows?

**Comment by joakime** [comment]
@olamy please be cognizant of the title of the PR and commit before you merge.

The `[WIP]` should not have been present in the merge message.
See commit ce09e48b5b29ec3165547bb9357cd15146c995d2

**Pull request merged**